src/foolscap/test/common.py

```diff
@@ -20,6 +22,28 @@
 from twisted.python import failure
 from twisted.internet.main import CONNECTION_DONE
 
+def convert_byte(x):
+    """ Convert to bytes type if x is a text type, else return as such """
+    # unicode -> str: Py2
+    # str -> bytes: Py3
+    if type(x) is six.text_type:
+        return six.ensure_binary(x)
+    return x
+
+def map_bytes(seq):
+    """ Map a sequence of strings to bytes type (recursive) """
```

Comment:
Why do we need this? It seems this is only used for this test, and so IMO it would be better to just replace all the string-literals with bytes-literals instead of running them through this method -- we have to replace all our "bare strings" with something anyway (i.e. everything should be `b""` or `u""` when the port is done).

Reply:
Too many places where there are nested data like ("string, ("foo", "x", ("y")) etc. This is a helper instead of manually modifying it in like some 50 places.

Reply:
I think the code will be a lot easier to read in the future w/o the new helpers -- we have to think about "should this be bytes or unicode" for every string anyway and using helpers just seems awkward instead of a search/replace.

Reply:
Ok, I was thinking the other way as searching for b'...' using your eyes IMHO is kind of difficult and the helper makes it explicit. Clearly we think differently.